tests/python/unittest/test_operator.py

```diff
@@ -7535,6 +7535,15 @@ def test_softmax():
     check_smoothed_softmax_grad(default_context())
 
 
+def create_data(batch_size, num_labels, ctx=None, dtype=None, H=None, W=None):
```

Comment:
Maybe you could reuse this test util https://github.com/apache/incubator-mxnet/blob/df34e768ed5d4dd334c0dce7ce42384d7b62ee8f/python/mxnet/test_utils.py#L92?

Reply:
Firts of all,
```
def random_arrays(*shapes):
    """Generate some random numpy arrays."""
    arrays = [np.random.randn(*s).astype(default_dtype())
              for s in shapes]
```
generates only `np.arrays`'s and we need `mx.nd.array`'s here.
Secondly, all elements of 'np.array' can only be of type `np.foat32`, and `dtype` is the input parameter of
```
def create_data( ... dtype=None, ...):
```
Third, the same approach is used to create data for 
```
def test_softmax_output_normalization():
 . . .
        if multi_output:
            data = create_data(batch_size, num_labels, ctx=None, H=H, W=W)
            label_shape = (batch_size, H, W)
        else:
            data = create_data(batch_size, num_labels)
            label_shape = (batch_size, )
```
I reused this one which, in my opinion is more versatile.